### PR TITLE
Autogenerated API Java client

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Autogenerate a type-safe java client to interact with the API. This is more ergonomic when hitting the API from java code and more importantly will cause Acceptance Tests to fail if the API contract is updated in backwards-incompatible way. 

## Reading Order
1. `dataline-api/build.gradle`
1. `AcceptanceTests.java`
1. `DatalineApiClient.java`
1. everything else

### CODE REVIEW

**Review by jrhizor — APPROVED** [review]

File: dataline-api/build.gradle [hunk 1]
```diff
@@ -1,11 +1,13 @@
 plugins {
-    id "org.openapi.generator" version "4.3.0"
+    id "org.openapi.generator" version "4.3.1"
     id "java-library"
 }
 
-def specFile = "$projectDir/src/main/openapi/config.yaml".toString()
+import org.openapitools.generator.gradle.plugin.tasks.GenerateTask;
```
Comment (michel-tricot) [inline dataline-api/build.gradle:6]: can you move it next to where you use it?

File: dataline-api/build.gradle [hunk 2]
```diff
@@ -15,41 +17,72 @@ openApiGenerate {
...
     implementation group: 'javax.validation', name: 'validation-api', version: '2.0.1.Final'
+
+    implementation group: 'org.openapitools', name: 'jackson-databind-nullable', version: '0.2.1'
 }
 
 sourceSets {
     main {
         java {
-            srcDir "$buildDir/generated/src/gen/java"
+            srcDirs "$buildDir/generated/src/gen/java", "$buildDir/generated/src/main/java", "$projectDir/src/main/java"
```
Comment (michel-tricot) [inline dataline-api/build.gradle:80]: can you put both the client and the sever code in the same generated/gen directory?

File: dataline-server/build.gradle [hunk 3]
```diff
@@ -7,10 +7,10 @@ dependencies {
     implementation group: 'org.eclipse.jetty', name: 'jetty-servlet', version: '9.4.31.v20200723'
 
     implementation group: 'org.glassfish.jaxb', name: 'jaxb-runtime', version: '2.3.3'
-    implementation group: 'org.glassfish.jersey.containers', name: 'jersey-container-servlet', version: '2.31'
-    implementation group: 'org.glassfish.jersey.inject', name: 'jersey-hk2', version: '2.31'
-    implementation group: 'org.glassfish.jersey.media', name: 'jersey-media-json-jackson', version: '2.31'
-    implementation group: 'org.glassfish.jersey.ext', name: 'jersey-bean-validation', version: '2.31'
+    implementation group: 'org.glassfish.jersey.containers', name: 'jersey-container-servlet'
```
Comment (michel-tricot) [inline dataline-server/build.gradle:10]: cool!

**Review by michel-tricot — APPROVED** [review]
love it

File: dataline-api/build.gradle (hunk 2 above)
Comment (sherifnada) [reply dataline-api/build.gradle:78]: this is generator specific unfortunately. The `java` generator puts them in the `main` directory

**Pull request merged**